pkg/admissioncontroller/server/handlers/webhooks/validate_namespace_deletion.go

```diff
@@ -41,20 +42,35 @@ import (
 type namespaceDeletionHandler struct {
 	k8sGardenClient kubernetes.Interface
 
-	projectLister gardencorelisters.ProjectLister
-	shootLister   gardencorelisters.ShootLister
-
 	codecs serializer.CodecFactory
 }
 
 // NewValidateNamespaceDeletionHandler creates a new handler for validating namespace deletions.
-func NewValidateNamespaceDeletionHandler(k8sGardenClient kubernetes.Interface, projectLister gardencorelisters.ProjectLister, shootLister gardencorelisters.ShootLister) http.HandlerFunc {
+func NewValidateNamespaceDeletionHandler(ctx context.Context, k8sGardenClient kubernetes.Interface) (http.HandlerFunc, error) {
+	// Initialize caches here to ensure http requests can be served quicker with pre-syncronized caches.
+	var hasSyncFuncs []cache.InformerSynced
+	informer, err := k8sGardenClient.Cache().GetInformerForKind(ctx, corev1.SchemeGroupVersion.WithKind("secret"))
```

Comment:
I think `Kind` should always be uppercase, right?
Anyways, can you use `GetInformer` with an API object to get type safety?
```suggestion
	informer, err := k8sGardenClient.Cache().GetInformer(ctx, &corev1.Secret{})
```